wp-content/themes/genesis-starter/functions.php

```diff
@@ -37,11 +37,20 @@
 
 //* Setup theme
 \add_action( 'after_setup_theme', function () {
-	$scripts    = new \Genesis_Starter\Scripts();
-	$styles     = new \Genesis_Starter\Styles();
-	$shortcodes = new \Genesis_Starter\Shortcodes();
-
-	$scripts->ready();
-	$styles->ready();
-	$shortcodes->ready();
+	$components = array(
+		'scripts'    => new \Genesis_Starter\Scripts(),
+		'shortcodes' => new \Genesis_Starter\Shortcodes(),
+		'styles'     => new \Genesis_Starter\Styles(),
+	);
+
+	// Excluded components
+	$exclude = \apply_filters( 'gs_exclude_components', array() );
+
+	foreach ( $components as $name => $instance ) {
+		if ( in_array( $name, $exclude, true ) ) {
+			continue;
+		}
+
+		$instance->ready();
```

Comment:
...although you should check that `$instance` implements `ready()` using `method_exists()`.

(You could force component classes to implement a common interface, but that creates cross-dependencies between theme and plugins that can crash the site if the theme is deactivated.)